### PR TITLE
Feature/lesson 19 typescript

### PR DESCRIPTION
**Pull request by anton-kan**

### CODE REVIEW

File: react-array-methods/package.json [hunk 1]
```diff
@@ -38,5 +38,14 @@
...
     ]
+  },
+  "devDependencies": {
+    "@types/jest": "^26.0.23",
+    "@types/node": "^16.0.0",
+    "@types/react": "^17.0.13",
+    "@types/react-dom": "^17.0.8",
+    "@types/react-router-dom": "^5.1.7",
+    "@types/styled-components": "^5.1.11",
+    "typescript": "^4.3.5"
```
Comment (bjatta) [inline react-array-methods/package.json:50]: redux?

File: react-array-methods/src/components/MethodsListsContainer/index.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import MethodsList from '../MethodsList';
+import Tooltip from '../Tooltip';
+import IMethodsList from './IMethodsList';
+
+import './index.css';
+
+interface IMethodsListsContainerProps {
+  readonly: boolean;
```
Comment (bjatta) [inline react-array-methods/src/components/MethodsListsContainer/index.tsx:8]: UpperCase I and Props in one time?
Here only?

File: react-array-methods/src/components/Counter.tsx [hunk 3]
```diff
@@ -1,7 +1,8 @@
-import React from 'react';
 import styled from 'styled-components';
 
 import { useSelector } from 'react-redux';
+import IState from '../store/IState';
+import ICounter from '../store/ICounter';
```
Comment (bjatta) [inline react-array-methods/src/components/Counter.tsx]: I prefer adding Props or Types at end of named TS decr.
Should it be more consistent for whole project?

File: react-array-methods/src/model/known-methods.ts [hunk 4]
```diff
@@ -1,29 +1,34 @@
...
   { method: 'keys', args: [] },
   { method: 'lastIndexOf', args: [3] },
-  { method: 'map', args: [(item) => item * 2] },
+  { method: 'map', args: [(item: number) => item * 2] },
   { method: 'pop', args: [] },
   { method: 'push', args: [15] },
-  { method: 'reduce', args: [(acc, value) => acc + value] },
-  { method: 'reduceRight', args: [(acc, value) => acc + value] },
+  { method: 'reduce', args: [(acc: number, value: number) => acc + value] },
+  { method: 'reduceRight', args: [(acc: number, value: number) => acc + value] },
```
Comment (bjatta) [inline react-array-methods/src/model/known-methods.ts]: may be it's an a time to separate aargis in 3 or 4 consts?

**Review by bjatta — APPROVED** [review]

File: react-array-methods/src/components/MethodsListsContainer/index.tsx (hunk 2 above)
Comment (anton-kan) [reply react-array-methods/src/components/MethodsListsContainer/index.tsx:8]: Not here only, I used upper-case I for all interfaces, regardless of whether they are Props or not

File: react-array-methods/src/components/Counter.tsx (hunk 3 above)
Comment (anton-kan) [reply react-array-methods/src/components/Counter.tsx]: Not sure I understand your comment... Can you please elaborate?

File: react-array-methods/src/model/known-methods.ts (hunk 4 above)
Comment (anton-kan) [reply react-array-methods/src/model/known-methods.ts]: Done

File: react-array-methods/package.json (hunk 1 above)
Comment (anton-kan) [reply react-array-methods/package.json:50]: https://redux-toolkit.js.org/tutorials/typescript

As of React Redux v7.2.3, the react-redux package has a dependency on @types/react-redux, so the type definitions will be automatically installed with the library.

**Pull request merged**